isisd: fix router capability TLV parsing issues

idryzhov opened the pull request:
This is a rebased version of https://github.com/FRRouting/frr/pull/9850.
A couple of adjustments made by me:
```
- fix incorrect replacement of srgb by srlb on lines 3052 and 3054
- add length check for ISIS_SUBTLV_ALGORITHM
- fix conflict in fuzzing data during rebase
```

whichbug commented on isisd/isis_tlvs.c:
```diff
@@ -3042,7 +3069,8 @@ static int unpack_tlv_router_cap(enum isis_tlv_context context,
+
 			break;
 		case ISIS_SUBTLV_ALGORITHM:
 			/* Only 2 algorithms are supported: SPF & Strict SPF */
```
We may miss a check on `length` before calling `stream_get`. We may need a check `length != 0` here.

idryzhov replied:
Fixed.

idryzhov commented:
I finally found and fixed the issue that caused CI failures. Added the following to the commit message:
```
Small adjustments by Igor Ryzhov:
- fix incorrect replacement of srgb by srlb on lines 3052 and 3054
- add length check for ISIS_SUBTLV_ALGORITHM
- fix conflict in fuzzing data during rebase

Signed-off-by: Igor Ryzhov <iryzhov@nfware.com>
```

riw777 reviewed (approved):
looks good

idryzhov commented:
@Mergifyio backport stable/8.2

idryzhov commented:
@Mergifyio backport dev/8.2
